src/main/scala/chisel3/experimental/util/SerializableModuleElaborator.scala

```diff
@@ -0,0 +1,80 @@
+package chisel3.experimental.util
+
+import chisel3.RawModule
+import chisel3.experimental.{SerializableModule, SerializableModuleGenerator, SerializableModuleParameter}
+
+import scala.reflect.runtime.universe
+import scala.reflect.runtime.universe.{runtimeMirror, typeOf}
+
+/** The target design emitted by [[SerializableModuleElaborator]], which contains
+  * @param fir the firrtl circuit
+  * @param annos annotations which is serializable
+  */
+case class SerializableModuleDesign(fir: firrtl.ir.Circuit, annos: Seq[firrtl.annotations.Annotation]) {
+  def firFile = fir.serialize
+  def annosFile = firrtl.annotations.JsonProtocol.serializeRecover(annos)
+}
```

Comment:
How about directly introducing the tuple for result? Adding new API here for only one usage might be over design, and I don't like to getting it deprecate in the future.

Reply:
I used it originally to provide unserialized structures to user. Now we return a tuple of geny.Readable hence we can use it in this way:
```scala
val (firrtl, annos) = designImpl[MySerializableModule, MySerializableModuleParameter](os.read.stream(parameter))
os.write.over(os.pwd / "GCD.fir", firrtl)
os.write.over(os.pwd / "GCD.anno.json", annos)
```